src/Microsoft.TestPlatform.ObjectModel/TestCase.cs

```diff
@@ -270,11 +268,7 @@ protected override void ProtectedSetPropertyValue(TestProperty property, object?
                 }
                 else if (value is string guidString)
                 {
-#if NET7_0_OR_GREATER
-                    Id = Guid.Parse(guidString, CultureInfo.InvariantCulture);
-#else
-                    Id = Guid.Parse(guidString);
-#endif
+                    Id = GuidPolyfill.Parse(guidString, CultureInfo.InvariantCulture);
```

Comment:
should the used signature been the "older" one, I mean from user perspective looks like the culture is impotant...but it's important only for one version...so I wonder if we should have the one without culture and setup the correct default for the newer one.

Reply:
That's a good suggestion. I don't see us using anything else than `InvariantCulture` so it would simplify calls. My idea here is to match newest API so that code looks like what you would expect if using recent .NET.

Maybe I will keep it as-is.